subprojects/language-native/src/main/java/org/gradle/language/rc/tasks/WindowsResourceCompile.java

```diff
@@ -201,15 +202,17 @@ public void setMacros(Map<String, String> macros) {
     }
 
     /**
-     * Additional arguments to provide to the compiler.
+     * <em>Additional</em> arguments to provide to the compiler.
+     *
+     * @since 5.1
      */
     @Input
-    public List<String> getCompilerArgs() {
+    public ListProperty<String> getCompilerArgs() {
         return compilerArgs;
     }
 
     public void setCompilerArgs(List<String> compilerArgs) {
```

Comment:
We are already changing the return type of the getter, I would just remove the setter and use the property everywhere it use the setter.

Reply:
I kept the setter in place because `WindowsResourceCompileTaskConfig` calls it when using the ``windows-resource'` plugin, and removing it would break things.  I'm not sure how to remove it without breaking the code for people still using deprecated functionality..

Reply:
Regarding removal of the setter, not sure what to do with `WindowsResourceCompileTaskConfig`'s use of the setter as it breaks the code here.  -> https://github.com/gradle/gradle/blob/817303166196b5bc7307400162bc40016921d53d/subprojects/language-native/src/main/java/org/gradle/language/rc/plugins/internal/WindowsResourcesCompileTaskConfig.java#L90   I'm in the process of creating a new pull request, and wasn't sure how to handle this issue.  Please advise.